### PR TITLE
Accept new field names from blackbox

### PR DESCRIPTION
**Pull request by McGiverGim**
The field names had been changing between versions. This commit adds a
"dictionary" to translate the names to the correct `sysConfig` parameter.

Please, review the matrix, because I'm new at the project and I don't know if I had mismatch the translation of some:
```
        translationValues = {                
            acc_limit_yaw             : "yawRateAccelLimit",
            accel_limit               : "rateAccelLimit",
            ac_limit                  : "rateAccelLimit",
            anti_gravity_thresh       : "anti_gravity_threshold",
            d_notch_cut               : "dterm_notch_cutoff", 
            d_setpoint_weight         : "dtermSetpointWeight",
            d_term_setpoint_weight    : "dtermSetpointWeight",  
            digital_idle_value        : "digitalIdleOffset",
            dshot_idle_value          : "digitalIdleOffset",
            gyro_lowpass              : "gyro_lowpass_hz",
            gyro_lowpass_type         : "gyro_lpf",
            "gyro.scale"              : "gyro_scale",
            iterm_windup              : "itermWindupPointPercent",
            motor_pwm_protocol        : "fast_pwm_protocol",
            pidsum_limit_yaw          : "yaw_p_limit",
            rc_expo                   : "rcExpo",
            rc_expo_yaw               : "rcYawExpo",
            rc_interp                 : "rc_interpolation",
            rc_interp_int             : "rc_interpolation_interval",
            rc_rate                   : "rcRate",
            rc_rate_yaw               : "rcYawRate",
            rc_yaw_expo               : "rcYawExpo",            
            setpoint_relax_ratio      : "setpointRelaxRatio",
            setpoint_relaxation_ratio : "setpointRelaxRatio",
            thr_expo                  : "thrExpo",
            thr_mid                   : "thrMid",
            tpa_rate                  : "dynThrPID",
            use_unsynced_pwm          : "unsynced_fast_pwm",
            vbat_scale                : "vbatscale",
            vbat_pid_gain             : "vbat_pid_compensation",
            yaw_accel_limit           : "yawRateAccelLimit"            	
    	},
```
Maybe in a future, we can add some metadata to this translations, to get rid of the big switch case to process each param.

### CODE REVIEW

File: js/flightlog_parser.js [hunk 1]
```diff
@@ -264,6 +264,41 @@ var FlightLogParser = function(logData) {
             unknownHeaders : []             // Unknown Extra Headers
         },
 
+        // Translation of the field values name to the sysConfig var where it must be stored
+        translationValues = {                
+            acc_limit_yaw             : "yawRateAccelLimit",
+            accel_limit               : "rateAccelLimit",
+            ac_limit                  : "rateAccelLimit",
```
Comment (martinbudden) [inline js/flightlog_parser.js]: `acc_limit`

File: js/flightlog_parser.js [hunk 2]
```diff
@@ -264,6 +264,41 @@ var FlightLogParser = function(logData) {
...
+            anti_gravity_thresh       : "anti_gravity_threshold",
+            d_notch_cut               : "dterm_notch_cutoff", 
+            d_setpoint_weight         : "dtermSetpointWeight",
+            d_term_setpoint_weight    : "dtermSetpointWeight",
```
Comment (martinbudden) [inline js/flightlog_parser.js]: `dterm_setpoint_weight`

**Comment by martinbudden** [comment]
Good stuff.

Personally I prefer the switch statement to the translation table, I think it's easier to maintain. But I'm not going to object to the PR on that basis.

Also, if you add the translate table, then you should remove the duplicated items from the switch statement.

File: js/flightlog_parser.js (hunk 1 above)
Comment (McGiverGim) [reply js/flightlog_parser.js]: Yes :-(

Changed.

File: js/flightlog_parser.js (hunk 2 above)
Comment (McGiverGim) [reply js/flightlog_parser.js]: Yes again :-(

Changed again.

**Comment by McGiverGim** [comment]
@martinbudden I haven't seen any duplicated items in the switch statement. I don't know if you have seen some. The others changes had been done, thanks for the revision.

Now that I have all the fields identified, I can remove the translation table and return to the switch statement if it's necessary, but the problem with the switch statement is that now, the values are grouped by behaviour. For example:
```
            case "rcRate":
            case "rcExpo":
            case "rcYawExpo":
            case "rcYawRate":
            ...
                that.sysConfig[fieldName] = parseInt(fieldValue, 10);
            break;
```
But if some of the elements have more than one "translation", the group is removed, generating a lot of code:
```
            case "rcRate":
            case "rcExpo":
            ...
                that.sysConfig[fieldName] = parseInt(fieldValue, 10);
            break;

            case "rcYawExpo":
            case "rc_expo_yaw":
                that.sysConfig["rcYawExpo"] = parseInt(fieldValue, 10);
            break;

            case "rcYawRate":
            case "rc_rate_yaw":
                that.sysConfig["rcYawRate"] = parseInt(fieldValue, 10);
            break;
```
This for each one of the translations of the table :-(

File: js/flightlog_parser.js [hunk 3]
```diff
@@ -487,7 +546,7 @@ var FlightLogParser = function(logData) {
                 } else {
                     that.sysConfig[fieldName] = parseInt(fieldValue, 10);
                 }
-                break;
```
Comment (mikeller) [inline js/flightlog_parser.js:490]: Indentation.

File: js/flightlog_parser.js [hunk 4]
```diff
@@ -477,7 +536,7 @@ var FlightLogParser = function(logData) {
             case "debug_mode":
                 that.sysConfig[fieldName] = parseInt(fieldValue, 10);
             break;
-
+
```
Comment (mikeller) [inline js/flightlog_parser.js]: Tab.

Comment (McGiverGim) [reply js/flightlog_parser.js]: Removed this and others similars.

File: js/flightlog_parser.js (hunk 3 above)
Comment (McGiverGim) [reply js/flightlog_parser.js:490]: Are you talking about the `break;`? The indentation is correct. This is the removed part and has been left as the others.

**Pull request merged**